liblouis/lou_translateString.c

```diff
@@ -3223,119 +3154,91 @@ beginCount(const EmphasisInfo *buffer, const int at, const EmphasisClass class,
 }
 
 static void
-insertEmphasesAt(const int at, const TranslationTableHeader *table, int pos,
-		const InString *input, OutString *output, int *posMapping,
-		const EmphasisInfo *emphasisBuffer, int haveEmphasis, int transOpcode,
+insertEmphasesAt(int begin, int end, int caps, int other, const int at,
+		const TranslationTableHeader *table, int pos, const InString *input,
+		OutString *output, int *posMapping, const EmphasisInfo *emphasisBuffer,
```

Comment:
This seems like a bit of a peculiar API. IIUC you have an argument `where` that can be `begin` or `end` and you have an argument `kind` that is either `caps` or `other`. You choose to have four bit arguments basically that map this API. Wouldn't it be better to have two enums?

A downside of the API you choose that in theory you would have to make sure the the function isn't called with both `begin` and `end` set.

Reply:
I agree the API isn't so nice, but there's no need for such checks. Any combination is allowed.

Reply:
I still think the API is flawed. However I looked at the code and all invocations do the right thing

Reply:
I would love a suggestion for improving the API of this function. But is it worth it?